test/acceptance/framework/flags/flags.go

```diff
@@ -44,25 +49,65 @@ func (t *TestFlags) init() {
 			"Note this flag must be run with -failfast flag, otherwise subsequent tests will fail.")
 	flag.StringVar(&t.flagECSClusterARN, flagECSClusterARN, "", "ECS Cluster ARN.")
 	flag.StringVar(&t.flagSubnets, flagSubnets, "", "Subnets to deploy into. In TF var form, e.g. '[\"sub1\",\"sub2\"]'.")
-	flag.StringVar(&t.flagSuffix, flagSuffix, "", "Resource suffix.")
 	flag.StringVar(&t.flagRegion, flagRegion, "", "Region.")
 	flag.StringVar(&t.flagLogGroupName, flagLogGroupName, "", "CloudWatch log group name.")
 	flag.StringVar(&t.flagTFTags, flagTFTags, "", "Tags to add to resources. In TF var form, e.g. '{key=val,key2=val2}'.")
+	flag.StringVar(&t.flagTFOutputDir, flagTFOutputDir, setupTerraformDir, "The directory of the setup terraform state for the tests.")
 }
 
 func (t *TestFlags) Validate() error {
 	// todo: require certain vars
 	return nil
 }
 
-func (t *TestFlags) TestConfigFromFlags() *config.TestConfig {
-	return &config.TestConfig{
-		NoCleanupOnFailure: t.flagNoCleanupOnFailure,
-		ECSClusterARN:      t.flagECSClusterARN,
-		Subnets:            t.flagSubnets,
-		Suffix:             t.flagSuffix,
-		Region:             t.flagRegion,
-		LogGroupName:       t.flagLogGroupName,
-		Tags:               t.flagTFTags,
+type tfOutputItem struct {
+	Value interface{}
+	Type  interface{}
+}
+
+func (t *TestFlags) TestConfigFromFlags() (*config.TestConfig, error) {
+	var cfg config.TestConfig
+
+	if t.flagTFOutputDir != "" {
+		var tfOutput map[string]tfOutputItem
+		testConfigMap := make(map[string]interface{})
+		cmd := exec.Command("terraform", "output", "-state", fmt.Sprintf("%s/terraform.tfstate", t.flagTFOutputDir), "-json")
+		cmdOutput, err := cmd.CombinedOutput()
+		if err != nil {
+			return nil, err
+		}
+		err = json.Unmarshal(cmdOutput, &tfOutput)
+		if err != nil {
+			return nil, err
+		}
+
+		for k, v := range tfOutput {
+			if k == "private_subnets" {
```

Comment:
comment about why subnets need to be handled specially would be nice

Reply:
I've added more comments to this code and also switched the output to be just 'subnets' in `setup-terraform`. I think it makes it easier for tests and since that TF is only used by tests I think it's OK.